tests/bluetooth/shell/testcase.yaml

```
@@ -43,6 +43,13 @@ tests:
     platform_allow:
       - native_sim
     build_only: true
+  bluetooth.shell.channel_sounding:
+    extra_configs:
+      - CONFIG_BT_CHANNEL_SOUNDING=y
+      - CONFIG_BT_CTLR=n
+    platform_allow:
+      - native_posix
```

Please use `native_sim`. `native_posix` is deprecated: https://docs.zephyrproject.org/latest/boards/native/native_posix/doc/index.html

```suggestion
      - native_sim
```